python/tank/platform/interactive_authentication.py

```diff
@@ -14,172 +14,272 @@
 
 from getpass import getpass
 import logging
+import threading
 
-from tank.errors import TankAuthenticationError
+from tank.errors import TankAuthenticationError, TankAuthenticationDisabled
 from tank.util import authentication
 from tank.util.login import get_login_name
 from tank.util import shotgun
 
 # Configure logging
 logger = logging.getLogger("sgtk.interactive_authentication")
```

Comment:
Do you feel like I should stub it out for now with an easy switch between the two if I need debugging?